seatunnel-translation/seatunnel-translation-base/src/main/java/org/apache/seatunnel/translation/source/ParallelSource.java

```diff
@@ -101,13 +101,13 @@ public ParallelSource(SeaTunnelSource<T, SplitT, StateT> source,
 
     @Override
     public void open() throws Exception {
+        parallelEnumeratorContext.register();
         executorService = ThreadPoolExecutorFactory.createScheduledThreadPoolExecutor(1, String.format("parallel-split-enumerator-executor-%s", subtaskId));
         splitEnumerator.open();
         if (restoredSplitState.size() > 0) {
             splitEnumerator.addSplitsBack(restoredSplitState, subtaskId);
         }
         reader.open();
-        parallelEnumeratorContext.register();
```

Comment:
why remove it?


@ashulin please help to review

Reply:
you should revert？

Reply:
why change it? 
I have the same question.